### ISSUE TITLE
Perfect scores reported for some models

### ISSUE THREAD

**Issue by torronen** [dotnet/machinelearning#5919]
In an experiment to evaluate AutoML performance and find ways to improve it we've trained about 180 000+ models with different AutoML settings and stored the metrics. We have 7 models out of 180 000 which report perfect metrics for both validation and test data. Datasets are fairly large, total of 4.5 gb, of which 1.5gb is validation and about 1 gb is test. However, we fail trying to "manually" reproduce the metrics by predicting the test set.

All of the 7 are trees (FastForestBinary, LightGbmBinary, FastTreeBinary) and in every instance the first run of AutoML had produced the perfect score. Running AutoML (through API) again with same settings does not anymore result in perfect scores (or even near). Unfortunately, we did not store the original zip files: we only have stored the re-fitted version (refitted with all data). I imagine refitting could alter the results a bit, because the almost half of the data was used for validation or testing.

It seems the issue can not be easily reproduced and the model zip file in question do not return predict scores even with test data. 

Dataset is on a network drive. For convenience, some of the nodes running AutoML were using Wifi, which could have caused connectivity issues. For example, could there be some cases where metrics are reported as perfect scores due to some sort of exceptions happening during the execution?

It is not a big issue, but is there something that could we should log to help find the root cause of this rare issue? 
For now, we can work around by re-running all experiments which only tried one model and produced perfect scores.

**Comment by torronen** [comment]
Suspect: Columns not ignored every time. They are ignore currently with:
`ColumnInference.ColumnInformation.IgnoredColumnNames.Add("ColumnToIgnore"); `
And also removed from all the other lists of column names (numeric, text ...)

It seems it is not only a problem with the perfect scores, but also most of the high-performing scores in the test.
It is probably a bug in my code. To confirm this is the only issue, we will re-run but dropping the columns instead of using the ignore function.
